src/Bicep.Core/Diagnostics/DiagnosticBuilder.cs

```diff
@@ -1782,20 +1782,10 @@ public ErrorDiagnostic NestedRuntimePropertyAccessNotSupported(string? resourceS
                 "BCP311",
                 $@"The provided index value of ""{indexSought}"" is not valid for type ""{typeName}"". Indexes for this type must be between 0 and {tupleLength - 1}");
 
-            public ErrorDiagnostic CyclicArmTypeRefs(IEnumerable<string> cycleLinks) => new(
+            public ErrorDiagnostic UnresolvableArmJsonType(string errorSource, string message) => new(
                 TextSpan,
                 "BCP312",
-                $@"Type reference cycle detected in the linked ARM JSON file (""{string.Join(" -> ", cycleLinks.Select(l => $@"""{l}"""))}"").");
-
-            public ErrorDiagnostic ArmTypeRefTargetNotFound(string refSansTarget) => new(
-                TextSpan,
-                "BCP313",
-                $@"The ""{refSansTarget}"" type definition could not be found in the linked ARM JSON file.");
-
-            public ErrorDiagnostic ArmTypeDefHasNoTypeOrRef(string invalidTypeDef) => new(
-                TextSpan,
-                "BCP314",
-                $@"The ""{invalidTypeDef}"" type definition in the linked ARM JSON file is invalid because it has neither a ""type"" property nor a ""$ref"" property.");
+                $@"The type at '{errorSource}' could not be resolved by the ARM JSON template engine. Original error message: [{message}]");
```

Comment:
Please could you be consistent with the quoting format we use in other diagnostics (double quotes):
```suggestion
                $@"The type at ""{errorSource}"" could not be resolved by the ARM JSON template engine. Original error message: ""{message}""");
```

For example: https://github.com/Azure/bicep/blob/84d8c5528902af301cef00724ef908027217f57c/src/Bicep.Core/Diagnostics/DiagnosticBuilder.cs#L1580